src/fides/api/input_validation.py

```diff
@@ -0,0 +1,56 @@
+"""Logic related to sanitizing and validating user application input."""
+from html import escape
+from re import compile as regex
+from typing import Generator
+
+
+class InputStr(str):
+    """
+    This class is designed to be used in place of the `str` type
+    any place where user input is expected.
+
+    The validation applied here does the typical sanitization/cleanup
+    that is required when dealing with user-supplied input.
+    """
+
+    @classmethod
+    def __get_validators__(cls) -> Generator:  # pragma: no cover
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: str) -> str:
+
+        # HTML Escapes
+        value = escape(value)
+
+        # Truncate to 500 characters
+        value = value[:500]
```

Comment:
This might surprise us a year from now - can we add an INFO log like `Truncated input string for field 'username' to safe maximum of 500 characters`?

(I'm hoping it might be possible to get the field name from the pydantic `cls` but that might be wishful thinking, if not then at least the breadcrumb trail reminding us of the existence of `InputStr`...)

Reply:
Like you said, this feels like an absolute pain of a bug to hunt down in 1 year...I'm going to hard fail instead. Silently (or even logging) a truncation doesn't feel right